### DISCUSSION TITLE
is it possible to Get page content id "ContentItemid" from the child part

### DISCUSSION THREAD

**Discussion by yassinsayed** [OrchardCMS/OrchardCore#11428] [Q&A]
the below will retrieve all child parts in content (page) 
var CEWidgets = part.ContentItem.Content.FlowPart.Widgets;
But I need to get the contentitemid of the page in the child Part or filed 

I add Form content and I add inputpart to the form 
I need to retrieve the form id from inside the inputpart

really need help

**Comment by Skrypt** [comment]
One trick is to pass the ContentItemId as a Shape Metadata inside your nested views hierarchy. Something like : 

Model.Content.MetaData.ParentContentItemId = someuuid

Model.Content is a Dynamic object ...

Else Shapes also have MetaData.

**Reply by ns8482e** [reply]
@yassinsayed where do you need it ? In drivers/views?

**Reply by yassinsayed** [reply]
In Driver 
It seems you read my thoughts

```C#
public override IDisplayResult Edit(InputPart part)
        {
            var **ParentId** = part.ContentItem.Content.Form.Id;
// need to retrieve form attribute here then code to do

            return Initialize<InputPartEditViewModel>("InputPart_Fields_Edit", m =>
            { 
                m.Placeholder = part.Placeholder;
                m.DefaultValue = part.DefaultValue;
                m.Type = part.Type;                
            });
        }
```

**Reply by ns8482e** [reply]
You can use HttpContext.Items or HttpContext.Features to set Parent’s ContentItemId using IContentsDriver Implementation of  parent content type

**Reply by Skrypt** [reply]
Not a bad idea too.

**Reply by jtkech** [reply]
Just for info the ShellScope also have Items and Features helpers.

Hmm, but this would not work if you are rendering multiple parents in the same scope.

What I'm also used to do in a parent shape is to use the child IShape.Properties dictionary and its extension helpers.

That said, maybe I misunderstood the thread but if you just need in a given part to retrieve the parent ContentItemId, normally you just need to do

    myPart.ContentItem.ContentItemId
